### DISCUSSION TITLE
How do I respond to filter changes?

### DISCUSSION THREAD

**Discussion by kirochoneski** [ghiscoding/Angular-Slickgrid#902] [Q&A] [answered]
We're migrating from v2 to v4, and noticed that gridInstance.filterService.onFilterChanged is no more. We were using it to respond to filter changes.

I tried following the changes to find a deprecation notice, but when switching to @slickgrid-universal/common repo, it is gone completely.

How can I listen to filter changes event in latest version?

**Comment by ghiscoding** [comment, marked as answer]
The `onFilterChanged` is a SlickGrid event, so you can call it directly from the View just like you could always do in the past, see this [example](https://github.com/ghiscoding/Angular-Slickgrid/blob/master/src/app/examples/grid-tree-data-parent-child.component.html#L97) or you could also use [Grid State - Wiki](https://github.com/ghiscoding/Angular-Slickgrid/wiki/Grid-State-&-Preset#grid-state-event) for filter and many other changes as well
